### DISCUSSION TITLE
TestNets supporting v2

### DISCUSSION THREAD

**Discussion by nsaun-dq** [WalletConnect/walletconnect-monorepo#2438] [Developer Support] [answered]
hey, I have completely tested my wallet against the demo app provided by you guys for v2, but I want to test it on some real websites, testnets will do too. Is there any way I can test it against opensea or rarible. My use case is to verify if it is working fine for "approve" kind of transactions

**Comment by chendinh** [comment]
i think nobody want to share project's testnet domain to test your transactions. 
actually demo app v2 which walletconnect team provided is good enough for you. I still use it for testing new walletconnect version.

**Reply by nsaun-dq** [reply]
I have test it against the demo app v2 and it is working completely fine. but just to be sure, I wanted to test it against some actual websites like rarible, opensea. Both of them provide testnets but the issue is the testnet dapps also only suuport v1 for now. I got a doc from @finessevanes for wallet devs, where the rarible link provided was still supporting v1 and that @finessevanes will talk to rarible team and let me know, but we moved from discord to here.

**Reply by chendinh** [reply]
https://demo.samuraiguild.io/
You can test Sign message transaction first on my demo web.
after you connect v2 to dapp, we need you sign message to authen you are real user

**Reply by chendinh** [reply]
If you want test send transaction on avax testnet. I can help you. I have order demo app. Give me your test address, i will send you some avax-testnet for gas fee and you just click subscribe button that will accept our system use your usdc-avax testnet.

**Comment by finessevanes** [comment, marked as answer]
@nsaun-dq https://docs.walletconnect.com/2.0/advanced/migration-from-v1.x/wallets/wallet-checklist#production-dapps!

We've moved the staging links here. Sorry, our conversation was in the middle of our GitHub move. Thanks for reposting your question here.
